cape-ph/cape-cod
# 46 implement push based queuing for etl

TO TEST
-------------
* Destroy/rebuild the deployment to get the best starting state. If you do not do this, ensure the HAI raw/clean buckets are empty and so is the SQS queue for hai
* Gather a data set. For my testing i mad a `gphl-cre` directory containing 6 copies of the docx and a `tnl` directory containin 10 copies of the xlsx. (***NOTE*** you want at least 6 of each file. we have 5 available glue runners for any job and you want to make sure to use all of them and see requeuing of messages happen when there are no resources available.
* upload this data set to the raw hai s3 bucket
* got to the cloud watch logs for the sqs trigger function (`ccd-dlh-T-hai-sqslmbdtrgfnct-XXXXXX` where `XXXXXX` is random chars) and ensure that you see some successes (for 5 of any job handling things) and exceptions for any that do not have available resources.
* feel free to poke around and examine anything else you want, but ultimately all files should be processed by glue and a csv file should end up in the clean bucket for it
